BenchmarkDotNet/BenchmarkRunner.cs

```diff
@@ -295,5 +290,76 @@ private static void AssertBenchmarkMethodIsNotGeneric(MethodInfo methodInfo)
             if (methodInfo.IsGenericMethod)
                 throw new InvalidOperationException($"Benchmark method {methodInfo.Name} is generic.\nGeneric benchmark methods are not supported.");
         }
+
+        /// <summary>
+        /// Given a list of benchmark statistics creates a function to convert 
+        /// raw <see cref="AverageTime"/> measurements to string format so that they
+        /// are shown using uniform time units and align nicely.
+        /// The <see cref="AverageTime"/> measurements are assumed to contain time lengths in nanoseconds.
+        /// </summary>
+        /// <param name="statistics">The list of time-based <see cref="BenchmarkRunReportsStatistic"/>.</param>
+        /// <returns>A function which should be used to convert all <see cref="AverageTime"/> measurements to string.</returns>
+        /// <remarks>
+        /// The measurements are formatted in such a way that they use the same time unit 
+        /// the number of decimals so that they are easily comparable and align nicely.
+        /// 
+        /// Example:
+        /// Consider we have the following raw input where numbers are durations in nanoseconds: 
+        ///     Median=597855, StdErr=485;
+        ///     Median=7643, StdErr=87;
+        /// 
+        /// When using the formatting function, the output will be like this:
+        ///     597.8550 us, 0.0485 us;
+        ///       7.6430 us, 0.0087 us;
+        /// </remarks>
+        public static Func<double, string> GetTimeMeasurementFormattingFunc(IEnumerable<BenchmarkRunReportsStatistic> statistics)
+        {
+            // Find the smallest measurement in the primary statistics, which is the Median.
+            // This will determine the time unit we will use for all measurements.
+            var minRecordedMedian = statistics.Min(stat => stat.AverageTime.Median);
+
+            // Use the largest unit to display the smallest recorded measurement without loss of precision.
+            // TODO: This is duplicated in BenchmarkMeasurementStatistic.ToString(), figure out how to refactor this later?
+            Func<double, string> measurementToString;
+            if (minRecordedMedian < 1000)
```

Comment:
OK, we could do it in several ways, but I am not convinced any of it makes things any clearer or better.

Using `for` loop:
```
const int baseScale = 1000;
var timeUnits = new[] { "ns", "us", "ms", " s" };

int selectedTimeUnitIndex;
for (selectedTimeUnitIndex = 0;
    selectedTimeUnitIndex < timeUnits.Length; 
    selectedTimeUnitIndex++, minRecordedMedian /= baseScale)
{
    if (minRecordedMedian < baseScale) break;
}

var adjustedScale = (int)Math.Pow(baseScale, selectedTimeUnitIndex);
var adjustedTimeUnit = timeUnits[selectedTimeUnitIndex];
return (value) => string.Format(EnvironmentHelper.MainCultureInfo, "{0:N4} {1}", value / adjustedScale, adjustedTimeUnit);
```

A more fun(c) way:
```
const int baseScale = 1000;
var timeUnits = 
    new[] { "ns", "us", "ms", " s" }
    .Select((timeUnit, index) =>
    {
        var adjustedScale = Math.Pow(baseScale, index);
        var isPicked = minRecordedMedian / adjustedScale < baseScale;
        return new { timeUnit, adjustedScale, picked = isPicked, };
    });

var selected = timeUnits.FirstOrDefault(t => t.picked) ?? timeUnits.Last();
return (value) => string.Format(EnvironmentHelper.MainCultureInfo, "{0:N4} {1}", value / selected.adjustedScale, selected.timeUnit);
```

Unless I'm doing something completely stupid here?